src/main/java/com/github/ogomezso/kafka/connect/soap/client/SoapClient.java

```diff
@@ -0,0 +1,105 @@
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.github.ogomezso.kafka.connect.soap.client;
+
+import javax.xml.namespace.QName;
+import javax.xml.transform.stream.StreamSource;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.time.LocalDateTime;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import com.github.jcustenborder.kafka.connect.utils.config.ConfigUtils;
+import com.github.ogomezso.kafka.connect.soap.source.SoapSourceConnectorConfig;
+
+import jakarta.xml.soap.MessageFactory;
+import jakarta.xml.soap.SOAPException;
+import jakarta.xml.soap.SOAPMessage;
+import jakarta.xml.soap.SOAPPart;
+import jakarta.xml.ws.Dispatch;
+import jakarta.xml.ws.Service;
+import jakarta.xml.ws.soap.SOAPBinding;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class SoapClient {
+
+  private final ScheduledExecutorService executor = Executors.newScheduledThreadPool(4);
+
+  private Callable<SOAPMessage> task;
+
+
+  public void start(SoapSourceConnectorConfig config) {
+    QName serviceName = new QName(config.getString(SoapSourceConnectorConfig.TARGET_NAMESPACE),
+        config.getString(SoapSourceConnectorConfig.SERVICE_NAME));
+    QName portName = new QName(config.getString(SoapSourceConnectorConfig.TARGET_NAMESPACE),
+        config.getString(SoapSourceConnectorConfig.PORT_NAME));
+    String endpointUrl = config.getString(SoapSourceConnectorConfig.ENDPOINT_URL);
+    String actionUrl = config.getString(SoapSourceConnectorConfig.SOAP_ACTION);
+    File messageFile = ConfigUtils
+        .getAbsoluteFile(config, SoapSourceConnectorConfig.REQUEST_MSG_FILE);
+
+    this.task = () -> invoke(serviceName, portName, endpointUrl, actionUrl,
+        messageFile);
+  }
+
+  public SOAPMessage poll(Long pollInterval)
+      throws ExecutionException, InterruptedException, TimeoutException {
+    ScheduledFuture<SOAPMessage> future = executor
+        .schedule(task, pollInterval,
+            TimeUnit.SECONDS);
+
+    log.debug("invonking at: " + LocalDateTime.now());
+    SOAPMessage result = future
+        .get(pollInterval + 5L,
+            TimeUnit.SECONDS);
+    log.debug("result: " + result.toString());
+    log.debug("get future at : " + LocalDateTime.now());
+    return result;
+  }
+
+  private SOAPMessage invoke(QName serviceName, QName portName, String endpointUrl,
+      String soapActionUri, File pathToMessage) throws SOAPException, FileNotFoundException {
+    Service service = Service.create(serviceName);
```

Comment:
are we creating a new connection every time we poll? If so maybe consider some resource pooling over keep alive connections.

Reply:
On TODO list for further PR (I'm gonna use Apache Commons Object pool as we discussed)